### PR TITLE
feat: pinned repositories/organizations on profile with recycler view…

### PR DESCRIPTION
**Pull request by fknop**
…/card

screenshot_2017-06-01-00-10-23

### CODE REVIEW

File: app/src/main/java/gitlin/kothub/adapters/OrganizationSummaryAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,47 @@
+package gitlin.kothub.adapters
+
+import android.content.Context
+import android.graphics.Color
```
Comment (Astalaseven) [inline app/src/main/java/gitlin/kothub/adapters/OrganizationSummaryAdapter.kt]: Remove useless imports

File: app/src/main/java/gitlin/kothub/adapters/PinnedRepositoryAdapter.kt [hunk 2]
```diff
@@ -1,45 +1,68 @@
...
+            vh.languageColor.background = gradient
+        }
+        
+        vh.forks.value = repository.forks
+        vh.stargazers.value = repository.stargazers
     }
 
+
+    class ViewHolder(view: View): RecyclerView.ViewHolder(view) {
+        val name = view.repository
```
Comment (Astalaseven) [inline app/src/main/java/gitlin/kothub/adapters/PinnedRepositoryAdapter.kt]: Add type explicitly?

File: app/src/main/java/gitlin/kothub/github/api/Base.kt [hunk 3]
```diff
@@ -2,40 +2,45 @@ package gitlin.kothub.github.api
 
 import android.util.Log
 import com.github.kittinunf.fuel.Fuel
-import com.github.kittinunf.fuel.android.extension.responseJson
 import com.github.kittinunf.fuel.core.FuelError
 import com.github.kittinunf.fuel.core.Request
 import com.github.kittinunf.result.Result
+import com.github.salomonbrys.kotson.get
+import com.github.salomonbrys.kotson.obj
```
Comment (Astalaseven) [inline app/src/main/java/gitlin/kothub/github/api/Base.kt]: Remove useless import

File: app/src/main/java/gitlin/kothub/github/api/Base.kt [hunk 4]
```diff
@@ -2,40 +2,45 @@ package gitlin.kothub.github.api
...
                 is Result.Success -> {
-                    val obj = result.value.obj()
-                    if (obj.isNull("data")) {
+
+                    val json = JsonParser().parse(result.value)
+                    val data = json["data"]
+
+                    if (data.isJsonNull) {
+                        val errors = json["errors"]
+                        // TODO: return errors
```
Comment (Astalaseven) [inline app/src/main/java/gitlin/kothub/github/api/Base.kt:37]: Todo to do?

File: app/src/main/res/layout/organization_view_list.xml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+            <TextView
+                android:id="@+id/organization"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:textAlignment="center"
+                android:layout_centerHorizontal="true"
+                android:ellipsize="end"
+                android:maxLines="1"
+                android:textSize="16dip"
```
Comment (Astalaseven) [inline app/src/main/res/layout/organization_view_list.xml]: Use `sp` instead

File: app/src/main/res/layout/repository_list_view.xml [hunk 6]
```diff
@@ -1,25 +1,117 @@
...
+        android:orientation="vertical"
+        android:padding="6dp">
+
+        <TextView
+            android:id="@+id/repository"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:ellipsize="end"
+            android:maxLines="1"
+            android:textSize="16dip"
```
Comment (Astalaseven) [inline app/src/main/res/layout/repository_list_view.xml]: Use `sp` for textSize

File: app/src/main/res/layout/fragment_user_profile.xml [hunk 7]
```diff
@@ -0,0 +1,196 @@
...
+    <TextView
+        android:id="@+id/pinnedTitle"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginEnd="@dimen/activity_horizontal_margin"
+        android:layout_marginLeft="@dimen/activity_horizontal_margin"
+        android:layout_marginRight="@dimen/activity_horizontal_margin"
+        android:layout_marginStart="@dimen/activity_horizontal_margin"
+        android:layout_marginTop="16dp"
+        android:text="Pinned Repositories"
```
Comment (Astalaseven) [inline app/src/main/res/layout/fragment_user_profile.xml]: Move text to `strings.xml`

File: app/src/main/res/layout/fragment_user_profile.xml [hunk 8]
```diff
@@ -0,0 +1,196 @@
...
+    <TextView
+        android:id="@+id/organizationsTitle"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginEnd="@dimen/activity_horizontal_margin"
+        android:layout_marginLeft="@dimen/activity_horizontal_margin"
+        android:layout_marginRight="@dimen/activity_horizontal_margin"
+        android:layout_marginStart="@dimen/activity_horizontal_margin"
+        android:layout_marginTop="16dp"
+        android:text="Organizations"
```
Comment (Astalaseven) [inline app/src/main/res/layout/fragment_user_profile.xml]: Move text to `strings.xml`

**Comment by Astalaseven** [comment]
New organizations don't appear. Can we grant them without having to log out?

**Comment by fknop** [comment]
Probably not, there is no API to authorize that. The easiest way is to grant it on GitHub site.

**Review by Astalaseven — CHANGES_REQUESTED** [review]

**Review by Astalaseven — APPROVED** [review]

**Pull request merged**